### DISCUSSION TITLE
이번 멘토링 코드리뷰 관련

### DISCUSSION THREAD

**Discussion by fkdl0048** [fkdl0048/Mentoring-Project#26] [Q&A] [answered]
코드리뷰를 한번도 제대로 받아본 적이 없어서 코드를 리뷰받는 시간을 한번 가져보고 싶은데 이번 멘토링에서 가능할까요?

미리 준비해야 하는게 있다면 미리 준비해보겠습니다..!

**Comment by jongfeel** [comment, marked as answer]
미리 어떤 코드에 대한 리뷰인지 알려주시면 더 좋을 수 있습니다.

**Reply by fkdl0048** [reply]
전에 보여드린 몬스터 클래스 관련 코드입니다.

fsm모델로 행동을 한가지 클래스로 정의하고 해당 행동머신을 따로 만들어서 행동을 진입, 반복, 퇴장으로 관리합니다.
따라서 한가지 몬스터가 있다고 했을 때 가질 수 있는 행동을 미리 정의하고 해당 몬스터가 정의된 행동을 namespace공간에 행동으로 정의해두었습니다. 

따라서 몬스터 클래스는 내부에 행동 클래스 state를 상속받은 행동 배열과 머신을 가지고 동작하게 만들어 두었습니다.
이를 한번에 관리, 간결하게 하기 위해 몬스터 부모 클래스를 만들고 공통된 부분을 추려놓았는데 경계를 잡기가 애매합니다..

++ OwendStates안에 동작부분에 전부 동작에 대한 정의를 하고 싶은데 그게 무리가 있는 것 같아서 몬스터 클래스에 일단 몰아 넣었습니다..

리뷰때 좀 더 자세히 말씀드리겠습니다 순서도 첨부합니다!

**Reply by fkdl0048** [reply]
![MonsterDrawIo](https://user-images.githubusercontent.com/84510455/207013744-f1521a78-27f2-407b-aaad-c969f3402b0e.png)
